libraries/AP_Proximity/AP_Proximity_LightWareSerial.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include "AP_Proximity.h"
+#include "AP_Proximity_Backend_Serial.h"
+
+#define PROXIMITY_LIGHTWARE_PAYLOAD_LEN_MAX 256 // maximum payload size we can accept (in some configurations sensor may send as large as 1023)
+
+class AP_Proximity_LightWareSerial : public AP_Proximity_Backend_Serial
+{
+
+public:
+    // constructor
+    using AP_Proximity_Backend_Serial::AP_Proximity_Backend_Serial;
+
+protected:
+
+    // initialise sensor
+    void initialise();
+
+    // send message to sensor
+    void send_message(uint8_t msgid, bool write, const uint8_t *payload, uint16_t payload_len);
+
+    // process one byte received on serial port
+    // returns true if a complete message has been received
+    // state is stored in _msg structure
+    bool parse_byte(uint8_t b);
+
+    enum class ParseState {
+        HEADER = 0,
+        FLAGS_L,
+        FLAGS_H,
+        MSG_ID,
+        PAYLOAD,
+        CRC_L,
+        CRC_H
+    };
+
+    // structure holding latest message contents
```

Comment:
Do we really need this `_msg` structure?  I'm not sure all the fields (e.g. crc_expected and the parse state) actually meet the description on the structure.

Reply:
ok, I've moved state, payload_recv and crc_expected out of the msg structure.